### ISSUE TITLE
Why is DenseMatrixBase::multiply not a static method?

### ISSUE THREAD

**Issue by roystgnr** [libMesh/libmesh#738]
It looks like it's a helper where the "this" matrix goes completely unused (except insofar as it's been passed in as M1, M2, or M3).  This isn't a polymorphic or a template specialized or any other crazy sort of method...

**Comment by jwpeterson** [comment]
```
git log -S"void DenseMatrixBase<T>::multiply" -- trunk/libmesh/src/numerics/dense_matrix_base.C
```

It dates back to at least 2003, when we added `DenseMatrixBase` as a common base class for both `DenseMatrix` and `DenseSubMatrix`, so I'm going to go with "didn't know about class static methods at the time".

**Comment by jwpeterson** [comment]
Fixed in 2856671.
